nixos/modules/services/misc/gerrit.nix

```
@@ -0,0 +1,158 @@
+
+{
+
+  options = {
+
+    services.gerrit = {
+
+      enable = mkEnableOption "Gerrit Code Review";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.gerrit;
+        defaultText = "pkgs.gerrit";
+        description = ''
+          The package used for the Gerrit daemon.
+        '';
+      };
+
+      settings = mkOption {
+        type = with types; lazyAttrsOf (lazyAttrsOf str);
+        default = {};
+      };
+
+      basePath = mkOption {
+        type = types.str;
+        default = "git";
+        description = ''
+          Local filesystem directory holding all Git repositories that Gerrit 
+          knows about and can process changes for. A project entity in Gerrit 
+          maps to a local Git repository by creating the path string 
+          "''${basePath}/''${project_name}.git".
+
+          If relative, the path is resolved relative to /var/lib/gerrit.
+        '';
+      };
+
+      canonicalWebUrl = mkOption {
+        type = types.str;
+        description = ''
+          The default URL for Gerrit to be accessed through.
+          Typically this would be set to something like 
+          "http://review.example.com/" or "http://example.com:8080/gerrit/" so 
+          Gerrit can output links that point back to itself.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "0.0.0.0:80";
+      };
+
+      plugins = mkOption {
+        type = with types; listOf path;
+        default = [];
+      };
+
+      pluginSettings = mkOption {
+        type = with types; lazyAttrsOf (lazyAttrsOf str);
+        default = {};
+      };
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    services.gerrit.settings = {
+      gerrit = {
+        inherit (cfg) basePath canonicalWebUrl;
+      };
+      httpd.inheritChannel = "true";
+      # TODO(edef): support SSH properly
+      sshd.listenAddress = "off";
+      cache.directory = "/var/cache/gerrit";
+    };
+
+    users.users.gerrit = {};
+
+    systemd.sockets.gerrit = {
+      unitConfig.Description = "Gerrit HTTP socket";
+      socketConfig.Accept = false;
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [ cfg.listenAddress ];
+    };
+
+    systemd.services.gerrit = {
+      unitConfig.Description = "Gerrit Code Review";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "gerrit.socket" ];
+      after = [ "gerrit.socket" "syslog.target" "network.target" ];
+
+      serviceConfig.Type = "simple";
+      serviceConfig.StandardInput = "socket";
+      serviceConfig.StandardOutput = "journal";
+
+      serviceConfig.User = "gerrit";
+      serviceConfig.CacheDirectory = "gerrit";
+      serviceConfig.StateDirectory = "gerrit";
+      serviceConfig.WorkingDirectory = "%S/gerrit";
+
+      environment.GERRIT_TMP = "%T";
+      serviceConfig.PrivateTmp = true;
+
+      serviceConfig.ExecStartPre = pkgs.writeScript "gerrit-pre" ''
+        #! ${pkgs.bash}/bin/bash -eu
```

a matter of preference
```suggestion
      serviceConfig.ExecStartPre = pkgs.writeShellScript "gerrit-pre" ''
        set -euo pipefail
```